### PR TITLE
docs(examples): add example working in new architecture

### LINKED ISSUE #189: docs(examples): cross-ledger CRUD in the new architecture
### Description

As a developer/contributor I want to have updated documentation with a proper, end to end example demonstrating how to develop applications with Cactus, so that I am not confused about the fact that (currently) there is no such documentation since we've moved over to the new architecture.

### Acceptance criteria

1. Demonstrates consortium with multiple members and their nodes
2. Demonstrates intra-node routing in action
3. Demonstrates cross ledger CRUD capabilities for a business logic plugin
4. SOLID principles are kept for the example app as well not just the core cactus code: https://en.wikipedia.org/wiki/SOLID
5. Short, deterministic script in the mean README.md file that can be executed by anyone to verify if the example is working.
6. Implement and leverage the keychain in-memory implementation (for demonstration purposes only)

### The original description of this issue before being refactored:

> ### Description
> As a developer/contributor I want to have updated documentation about the simple asset transfer demo that I saw during the Hyperledger Global Forum 2020 talk so that I am not confused about the fact that (currently) there is no such documentation since we've moved over to the new architecture.
> 
> Of course this depends on the new architecture reaching a level of maturity where the simple asset transfer case can actually be implemented.
> 
> ### Acceptance Criteria
> 1. The simple asset transfer use-case is re-written from scratch to match the new arch
> 2. The new documentation closes the loop regarding asset export verification and error handling as well (thanks to Arnaud for pointing out the lack of these)
> 
> cc: @hartm @sfuji822  @takeutak  @jonathan-m-hamilton
>

### PR DESCRIPTION
**Pull request by petermetz**
Fixes #189 

**Depends on #337 => meaning that #337 must be merged first prior to this getting merged.**

Huge PR that incorporates all the remaining changes/features necessary to have an actual working cross ledger asset transfer example (not an atomic swap, just a simple one).

It contains changes in several different packages that got piled up together to shorten the development time instead of slowly iterating with smaller but more digestible pull requests (a trade off made to move things along faster in the initial phases of development where we don't yet have a stable release).

The interesting changes are in the last commit that actually adds the example, the older commits of the PR are not specific to the example code but had to be put there because the example code depends on them.

Recommended to start reviewing by testing it first via the instructions placed in the README.md file.

The example does not feature asset transfers among ledgers (export) just shows how different contracts on different ledgers can be used to query and join data from different ledgers. This is a shortcut I had to take to be able to finish with the PR in reasonable time (well, it's still already way later than it should've been, but it is what it is).

Once you launch the example it will show you a web UI that you can click around on to create data on both ledgers that are involved with the example (quorum, besu).

Will need to add more documentation in subsequent commits and also update the whitepaper.

This PR contains the backbone of all the changes we need to push through with the v0.2.0 release (there are a few minor issues still left after this but those are just quick administrative ones that are related to documentation links and such).

cc: @takeutak @jonathan-m-hamilton @sfuji822

### CODE REVIEW

**Comment by petermetz** [comment]
Update: Marked it 'ready for review' and updated the PR description.
cc: @jonathan-m-hamilton @takeutak @sfuji822

**Comment by petermetz** [comment]
Also recommend to check out which files had more changes than others (not always a good indicator of *significance* but probably a good start)

```sh
$ git diff upstream/master HEAD --stat -- ':*.ts' ':!*.test.ts' ':!*/generated/*'
packages/cactus-api-client/src/main/typescript/api-client.ts                                                                         | 134 +++++++++++++++-
 packages/cactus-api-client/src/main/typescript/default-consortium-provider.ts                                                        |  65 ++++++++
 packages/cactus-api-client/src/main/typescript/public-api.ts                                                                         |   1 +
 packages/cactus-cmd-api-server/src/main/typescript/api-server.ts                                                                     |   7 +-
 packages/cactus-cmd-api-server/src/main/typescript/config/config-service.ts                                                          |  10 +-
 packages/cactus-cmd-api-server/src/main/typescript/public-api.ts                                                                     |   3 +
 packages/cactus-common/src/main/typescript/checks.ts                                                                                 |  19 +++
 packages/cactus-common/src/main/typescript/i-async-provider.ts                                                                       |  12 ++
 packages/cactus-common/src/main/typescript/index.ts                                                                                  |   6 +
 packages/cactus-common/src/main/typescript/nodejs/openapi-spec-json-to-fs.ts                                                         |  51 +++++++
 packages/cactus-common/src/main/typescript/public-api.ts                                                                             |   1 +
 packages/cactus-common/src/main/typescript/servers.ts                                                                                |  40 ++++-
 packages/cactus-core-api/src/main/typescript/openapi-spec.ts                                                                         | 331 ++++++++++++++++++++++++++++++++++++++++
 packages/cactus-core-api/src/main/typescript/plugin/i-cactus-plugin.ts                                                               |  75 +++++++--
 packages/cactus-core-api/src/main/typescript/plugin/keychain/i-plugin-keychain.ts                                                    |  14 ++
 packages/cactus-core-api/src/main/typescript/plugin/ledger-connector/i-plugin-ledger-connector.ts                                    |  20 ++-
 packages/cactus-core-api/src/main/typescript/plugin/plugin-registry.ts                                                               |  53 ++++---
 packages/cactus-core-api/src/main/typescript/plugin/web-service/i-plugin-web-service.ts                                              |   3 +-
 packages/cactus-core-api/src/main/typescript/plugin/web-service/i-web-service-endpoint.ts                                            |  32 ++++
 packages/cactus-core-api/src/main/typescript/public-api.ts                                                                           |   7 +-
 packages/cactus-core/src/main/typescript/index.ts                                                                                    |   1 +
 packages/cactus-core/src/main/typescript/index.web.ts                                                                                |   1 +
 packages/cactus-core/src/main/typescript/public-api.ts                                                                               |   1 +
 packages/cactus-core/src/main/typescript/web-services/register-web-service-endpoint.ts                                               |  29 ++++
 packages/cactus-core/src/test/typescript/integration/api-surface.ts                                                                  |   7 +
 packages/cactus-core/src/test/typescript/unit/api-surface.ts                                                                         |   7 +
 packages/cactus-example-supply-chain-backend/src/main/typescript/index.ts                                                            |   1 +
 packages/cactus-example-supply-chain-backend/src/main/typescript/index.web.ts                                                        |   1 +
 packages/cactus-example-supply-chain-backend/src/main/typescript/infrastructure/supply-chain-app-dummy-infrastructure.ts             | 199 ++++++++++++++++++++++++
 packages/cactus-example-supply-chain-backend/src/main/typescript/public-api.ts                                                       |   2 +
 packages/cactus-example-supply-chain-backend/src/main/typescript/supply-chain-app-cli.ts                                             |  20 +++
 packages/cactus-example-supply-chain-backend/src/main/typescript/supply-chain-app.ts                                                 | 324 +++++++++++++++++++++++++++++++++++++++
 packages/cactus-example-supply-chain-backend/src/test/typescript/integration/api-surface.ts                                          |  10 ++
 packages/cactus-example-supply-chain-backend/src/test/typescript/unit/api-surface.ts                                                 |   7 +
 packages/cactus-example-supply-chain-business-logic-plugin/src/main/typescript/business-logic-plugin/supply-chain-cactus-plugin.ts   | 131 ++++++++++++++++
 .../src/main/typescript/business-logic-plugin/web-services/insert-bamboo-harvest-endpoint-constants.ts                               |   9 ++
 .../src/main/typescript/business-logic-plugin/web-services/insert-bamboo-harvest-endpoint.ts                                         | 110 +++++++++++++
 .../src/main/typescript/business-logic-plugin/web-services/insert-bookshelf-endpoint-constants.ts                                    |   8 +
 .../src/main/typescript/business-logic-plugin/web-services/insert-bookshelf-endpoint.ts                                              | 110 +++++++++++++
 .../src/main/typescript/business-logic-plugin/web-services/list-bamboo-harvest-endpoint-constants.ts                                 |   8 +
 .../src/main/typescript/business-logic-plugin/web-services/list-bamboo-harvest-endpoint.ts                                           | 112 ++++++++++++++
 .../src/main/typescript/business-logic-plugin/web-services/list-bookshelf-endpoint-constants.ts                                      |   8 +
 .../src/main/typescript/business-logic-plugin/web-services/list-bookshelf-endpoint.ts                                                | 112 ++++++++++++++
 packages/cactus-example-supply-chain-business-logic-plugin/src/main/typescript/i-eth-contract-deployment.ts                          |   5 +
 packages/cactus-example-supply-chain-business-logic-plugin/src/main/typescript/i-supply-chain-contract-deployment-info.ts            |   6 +
 packages/cactus-example-supply-chain-business-logic-plugin/src/main/typescript/index.ts                                              |   1 +
 packages/cactus-example-supply-chain-business-logic-plugin/src/main/typescript/index.web.ts                                          |   1 +
 packages/cactus-example-supply-chain-business-logic-plugin/src/main/typescript/model/converter/bamboo-harvest-converter.ts           |  37 +++++
 packages/cactus-example-supply-chain-business-logic-plugin/src/main/typescript/model/converter/bookshelf-converter.ts                |  33 ++++
 packages/cactus-example-supply-chain-business-logic-plugin/src/main/typescript/openapi-spec.ts                                       | 276 +++++++++++++++++++++++++++++++++
 packages/cactus-example-supply-chain-business-logic-plugin/src/main/typescript/public-api.ts                                         |   9 ++
 packages/cactus-example-supply-chain-business-logic-plugin/src/test/typescript/integration/api-surface.ts                            |  10 ++
 packages/cactus-example-supply-chain-business-logic-plugin/src/test/typescript/unit/api-surface.ts                                   |   7 +
 packages/cactus-example-supply-chain-frontend/src/app/app-routing.module.ts                                                          |  30 ++++
 packages/cactus-example-supply-chain-frontend/src/app/app.component.ts                                                               |  41 +++++
 packages/cactus-example-supply-chain-frontend/src/app/app.module.ts                                                                  |  54 +++++++
 packages/cactus-example-supply-chain-frontend/src/app/bamboo-harvest/bamboo-harvest-detail/bamboo-harvest-detail.page.ts             |  68 +++++++++
 packages/cactus-example-supply-chain-frontend/src/app/bamboo-harvest/bamboo-harvest-list/bamboo-harvest-list.page.ts                 |  95 ++++++++++++
 packages/cactus-example-supply-chain-frontend/src/app/bamboo-harvest/bamboo-harvest-routing.module.ts                                |  17 +++
 packages/cactus-example-supply-chain-frontend/src/app/bamboo-harvest/bamboo-harvest.module.ts                                        |  24 +++
 packages/cactus-example-supply-chain-frontend/src/app/bookshelf/bookshelf-detail/bookshelf-detail.page.ts                            |  92 +++++++++++
 packages/cactus-example-supply-chain-frontend/src/app/bookshelf/bookshelf-list/bookshelf-list.page.ts                                |  93 +++++++++++
 packages/cactus-example-supply-chain-frontend/src/app/bookshelf/bookshelf-routing.module.ts                                          |  17 +++
 packages/cactus-example-supply-chain-frontend/src/app/bookshelf/bookshelf.module.ts                                                  |  24 +++
 packages/cactus-example-supply-chain-frontend/src/app/common/page-heading/page-heading-component.module.ts                           |  14 ++
 packages/cactus-example-supply-chain-frontend/src/app/common/page-heading/page-heading.component.ts                                  |  11 ++
 packages/cactus-example-supply-chain-frontend/src/constants.ts                                                                       |   9 ++
 packages/cactus-example-supply-chain-frontend/src/environments/environment.prod.ts                                                   |   3 +
 packages/cactus-example-supply-chain-frontend/src/environments/environment.ts                                                        |  16 ++
 packages/cactus-example-supply-chain-frontend/src/main.ts                                                                            |  14 ++
 packages/cactus-example-supply-chain-frontend/src/polyfills.ts                                                                       |  65 ++++++++
 packages/cactus-example-supply-chain-frontend/src/zone-flags.ts                                                                      |   5 +
 packages/cactus-plugin-consortium-manual/src/main/typescript/consortium/get-consortium-jws-endpoint-constants.ts                     |  15 ++
 packages/cactus-plugin-consortium-manual/src/main/typescript/consortium/get-consortium-jws-endpoint-v1.ts                            |  38 +++--
 packages/cactus-plugin-consortium-manual/src/main/typescript/consortium/get-node-jws-endpoint-constants.ts                           |  15 ++
 packages/cactus-plugin-consortium-manual/src/main/typescript/consortium/get-node-jws-endpoint-v1.ts                                  |  27 +++-
 packages/cactus-plugin-consortium-manual/src/main/typescript/openapi-spec.ts                                                         | 223 ++-------------------------
 packages/cactus-plugin-consortium-manual/src/main/typescript/plugin-consortium-manual.ts                                             |  47 ++++--
 packages/cactus-plugin-keychain-memory/src/main/typescript/plugin-factory-keychain.ts                                                |  13 +-
 packages/cactus-plugin-keychain-memory/src/main/typescript/plugin-keychain-memory.ts                                                 |  77 +++++++---
 packages/cactus-plugin-keychain-memory/src/main/typescript/public-api.ts                                                             |   5 +-
 packages/cactus-plugin-kv-storage-memory/src/main/typescript/plugin-factory-kv-storage.ts                                            |   7 +-
 packages/cactus-plugin-kv-storage-memory/src/main/typescript/plugin-kv-storage-memory.ts                                             |  26 +++-
 packages/cactus-plugin-ledger-connector-besu/src/main/typescript/model-type-guards.ts                                                |  24 +++
 packages/cactus-plugin-ledger-connector-besu/src/main/typescript/openapi-spec.ts                                                     | 585 ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 packages/cactus-plugin-ledger-connector-besu/src/main/typescript/plugin-ledger-connector-besu.ts                                     | 388 +++++++++++++++++++++++++++++++++++++++++-----
 packages/cactus-plugin-ledger-connector-besu/src/main/typescript/public-api.ts                                                       |   2 +
 .../cactus-plugin-ledger-connector-besu/src/main/typescript/web-services/deploy-contract-solidity-bytecode-endpoint-constants.ts     |  16 ++
 packages/cactus-plugin-ledger-connector-besu/src/main/typescript/web-services/deploy-contract-solidity-bytecode-endpoint.ts          |  82 ++++++++++
 packages/cactus-plugin-ledger-connector-besu/src/main/typescript/web-services/invoke-contract-endpoint-constants.ts                  |  16 ++
 packages/cactus-plugin-ledger-connector-besu/src/main/typescript/web-services/invoke-contract-endpoint.ts                            |  79 ++++++++++
 packages/cactus-plugin-ledger-connector-besu/src/main/typescript/web-services/run-transaction-endpoint-constants.ts                  |  16 ++
 packages/cactus-plugin-ledger-connector-besu/src/main/typescript/web-services/run-transaction-endpoint.ts                            |  79 ++++++++++
 .../src/test/typescript/integration/plugin-ledger-connector-besu/deploy-contract/deploy-contract-from-json.ts                        |  41 -----
 packages/cactus-plugin-ledger-connector-fabric/src/main/typescript/deploy-contract-go-bin-endpoint-constants.ts                      |  15 ++
 packages/cactus-plugin-ledger-connector-fabric/src/main/typescript/deploy-contract-go-bin-endpoint-v1.ts                             |  21 ++-
 packages/cactus-plugin-ledger-connector-fabric/src/main/typescript/plugin-ledger-connector-fabric.ts                                 |  35 ++++-
 packages/cactus-plugin-ledger-connector-quorum/src/main/typescript/model-type-guards.ts                                              |  40 +++++
 packages/cactus-plugin-ledger-connector-quorum/src/main/typescript/openapi-spec.ts                                                   | 584 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 packages/cactus-plugin-ledger-connector-quorum/src/main/typescript/plugin-ledger-connector-quorum.ts                                 | 419 +++++++++++++++++++++++++++++++++++++-------------
 packages/cactus-plugin-ledger-connector-quorum/src/main/typescript/public-api.ts                                                     |   8 +-
 packages/cactus-plugin-ledger-connector-quorum/src/main/typescript/web-services/deploy-contract-endpoint.ts                          |  43 ------
 .../cactus-plugin-ledger-connector-quorum/src/main/typescript/web-services/deploy-contract-solidity-bytecode-endpoint-constants.ts   |  16 ++
 packages/cactus-plugin-ledger-connector-quorum/src/main/typescript/web-services/deploy-contract-solidity-bytecode-endpoint.ts        |  82 ++++++++++
 packages/cactus-plugin-ledger-connector-quorum/src/main/typescript/web-services/invoke-contract-endpoint-constants.ts                |  16 ++
 packages/cactus-plugin-ledger-connector-quorum/src/main/typescript/web-services/invoke-contract-endpoint.ts                          |  79 ++++++++++
 packages/cactus-plugin-ledger-connector-quorum/src/main/typescript/web-services/run-transaction-endpoint-constants.ts                |  16 ++
 packages/cactus-plugin-ledger-connector-quorum/src/main/typescript/web-services/run-transaction-endpoint.ts                          |  79 ++++++++++
 .../src/test/typescript/integration/plugin-ledger-connector-quorum/deploy-contract/deploy-contract-from-json.ts                      |  74 ---------
 packages/cactus-plugin-validator-besu/src/main/typescript/besu/sign-transaction-endpoint-constants.ts                                |  15 ++
 packages/cactus-plugin-validator-besu/src/main/typescript/besu/sign-transaction-endpoint-v1.ts                                       |  22 ++-
 packages/cactus-plugin-validator-besu/src/main/typescript/plugin-validator-besu.ts                                                   |  28 +++-
 packages/cactus-test-api-client/src/main/typescript/index.ts                                                                         |   1 +
 packages/cactus-test-api-client/src/main/typescript/index.web.ts                                                                     |   1 +
 packages/cactus-test-api-client/src/main/typescript/public-api.ts                                                                    |   1 +
 packages/cactus-test-api-client/src/test/typescript/integration/api-surface.ts                                                       |   7 +
 packages/cactus-test-api-client/src/test/typescript/unit/api-surface.ts                                                              |  10 ++
 .../src/test/typescript/integration/plugin-consortium-manual/security-isolation-via-api-server-ports.ts                              |  33 ++--
 .../src/test/typescript/integration/plugin-ledger-connector-quorum/deploy-contract/deploy-contract-via-web-service.ts                | 124 ---------------
 packages/cactus-test-tooling/src/main/typescript/besu/besu-test-ledger.ts                                                            |  59 +++++++
 packages/cactus-test-tooling/src/main/typescript/quorum/quorum-test-ledger.ts                                                        |  51 +++++++
 typings/async-exit-hook/index.d.ts                                                                                                   |  36 +++++
 122 files changed, 6114 insertions(+), 801 deletions(-)

```

File: examples/supply-chain-app/package.json
Comment (takeutak) [inline examples/supply-chain-app/package.json:1]: Please divide codes depending on sample application to examples directory, not only package.json?

Comment (petermetz) [reply examples/supply-chain-app/package.json:1]: > Could you mind if you divide codes depending on sample application to examples directory, not only package.json? This is the same comment as I said on contributors' meeting.

@takeutak It can be done, but it complicates the build to have it outside of the packages directory. Would you consider it separated based on the fact that the example packages literally have their folders prefixed with the word `example`?

**Review by jonathan-m-hamilton — APPROVED** [review]
LGTM!

**Comment by takeutak** [comment]
Moreover, I don't feel comfortable because the title of your PR is "docs" and this PR includes not only documents but also actual codes.  Would you mind if you divide your PR into docs(example) and feat(example)?

**Comment by petermetz** [comment]
> Moreover, I don't feel comfortable because the title of your PR is "docs" and this PR includes not only documents but also actual codes. Would you mind if you divide your PR into docs(example) and feat(example)?

@takeutak Done: #337

**Comment by takeutak** [comment]
Could you explain the reasoning for the modification of the README and the configuration file of the top directory?
I still see some code which is not example codes. Can you remove them and repost them as another PR?

**Comment by petermetz** [comment]
> Could you explain the reasoning for the modification of the README and the configuration file of the top directory?
> I still see some code which is not example codes. Can you remove them and repost them as another PR?

@takeutak Assuming we cleared this up on the maintainer's call, but do let me know if there's any other questions, I want to make sure all of them are answered!

File: README.md
Comment (takeutak) [inline README.md:41]: Would you mind if you edit this file as the following?

---
### Car Trade Example

- The guidance is [here](/examples/cartrade/README.md)
---

**Review by takeutak — APPROVED** [review]
Thanks for your update on example app, Peter!  I approved this PR (docs(examples): supply chain app example). 
However, before merging, I would like you to rewrite the part of car-trade app on the README as my comment as the above, @petermetz.
Moreover, I think there remain many improvements in example apps (such as car-trade app and supply-chain app). I have made an issue as Issue #385 , so I would be happy to discuss and develop it separately.

File: README.md
Comment (petermetz) [reply README.md:41]: Done https://github.com/hyperledger/cactus/commit/09814f2a7904c76b425507cb457690426cfcce8d

**Comment by petermetz** [comment]
> However, before merging, 

@takeutak We cannot merge this PR before the other PR that this depends on even if this one now is approved. I was hoping you'd review that other one as well.

> I would like you to rewrite the part of car-trade app on the README as my comment as the above, @petermetz.

@takeutak Done.

> Moreover, I think there remain many improvements in example apps (such as car-trade app and supply-chain app). 

There are definitely a lot of improvements in there, I agree.

> I have made an issue as Issue #385 , so I would be happy to discuss and develop it separately.

@takeutak Great!

**Comment by takeutak** [comment]
> @takeutak We cannot merge this PR before the other PR that this depends on even if this one now is approved. I was hoping you'd review that other one as well.

With your modifications, the README already reflects my modifications.  I agree with the merge of this pull-request.

**Comment by petermetz** [comment]
> With your modifications, the README already reflects my modifications. I agree with the merge of this pull-request.

@takeutak Thank you. Do you mean that you agree with all 43 commits[2], or just the top commit of this PR, the one specifically adding the example packages[1]?

[1]: 
```
docs(examples): supply chain app example …
09814f2
The idea is to have a very quickly testable demonstration
of how Cactus can be used to develop an (overly simplistic)
business use case that involves transacting over multiple
ledgers. This example does not yet provide an exhaustive
example on how to implement atomic swaps for example,
it is purely aimed at showing how code can be written
with the help of Cactus to perform basic tasks on
different ledgers.

Signed-off-by: Peter Somogyvari <peter.somogyvari@accenture.com>
```

[2]:

```
build(webpack): add JSDoc type hints to config …

@petermetz
petermetz committed 4 days ago
 
feat(common): add Servers.startOnPreferredPort(), Servers.startOnPort() …

@petermetz
petermetz committed 4 days ago
 
chore(typings): add async-exit-hook npm package typings …

@petermetz
petermetz committed 4 days ago
 
build(package-json): configure now runs install and dev build only …

@petermetz
petermetz committed 4 days ago
 
build(package-json): npm script to run tests quorum connector only …

@petermetz
petermetz committed 4 days ago
 
fix(cmd-api-server): add IPluginImport to public API surface of package …

@petermetz
petermetz committed 4 days ago
 
fix(cmd-api-server): plugin imports via static config (env,cli,file) …

@petermetz
petermetz committed 4 days ago
 
build(commitizen): limit header to 72 from 100 …

@petermetz
petermetz committed 4 days ago
 
chore: update @types/express to 4.17.8 …

@petermetz
petermetz committed 4 days ago
 
refactor(core-api): add transact() to connector, remove addPublicKey() …

@petermetz
petermetz committed 4 days ago
 
build(test-plugin-consortium-manual): fix bundle names in package.json …

@petermetz
petermetz committed 4 days ago
 
refactor(core-api): renames the plugin id getter to package name …

@petermetz
petermetz committed 4 days ago
 
test(common): experiment to fix flaky test case …

@petermetz
petermetz committed 4 days ago
 
feat(common): add OpenAPI spec exporter utility function …

@petermetz
petermetz committed 4 days ago
 
build(tsconfig-json): include es2019 in libs to have Array#flat() …

@petermetz
petermetz committed 4 days ago
 
fix(common): servers.startOnPort() error handling …

@petermetz
petermetz committed 4 days ago
 
feat(common): add IAsyncProvider interface definition …

@petermetz
petermetz committed 4 days ago
 
refactor: move consortium OpenAPI schema to core-api …

@petermetz
petermetz committed 4 days ago
 
feat(sdk): routing to nodes by ledger ID …

@petermetz
petermetz committed 4 days ago
 
chore(plugin-ledger-connector-besu): unused imports cleanup …

@petermetz
petermetz committed 4 days ago
 
feat(plugin-ledger-connector-besu): add log level to options …

@petermetz
petermetz committed 4 days ago
 
build: building_construction fix git-cz mismatch of possible commit message types …

@petermetz
petermetz committed 4 days ago
 
build(webpack): adds shebang loader …

@petermetz
petermetz committed 4 days ago
 
feat(core-api): add instanceId getter to ICactusPlugin …

@petermetz
petermetz committed 4 days ago
 
feat(common): add Checks#nonBlankString() utility …

@petermetz
petermetz committed 4 days ago
 
feat: guitar add method: QuorumTestLedger#getGenesisAccount() …

@petermetz
petermetz committed 4 days ago
 
feat(besu): guitar OpenAPI support => arbitrary TX, contract deployments …

@petermetz
petermetz committed 4 days ago
 
chore(besu): robot migrate from tap to tape …

@petermetz
petermetz committed 4 days ago
 
test(common): ring Bools.isBooleanStrict() …

@petermetz
petermetz committed 4 days ago
 
feat(core-api): guitar add IKeychainPlugin#getKeychainId() …

@petermetz
petermetz committed 4 days ago
 
build: building_construction add top level tape-promise dependency for async tests …

@petermetz
petermetz committed 4 days ago
 
refactor: bulb In-memory keychain getEncryptionAlgorithm() => null …

@petermetz
petermetz committed 4 days ago
 
test: ring add plugin-keychain-memory unit test …

@petermetz
petermetz committed 4 days ago
 
feat(quorum): guitar export model-type-guards to public API surface …

@petermetz
petermetz committed 4 days ago
 
feat(quorum): guitar support Cactus Keychain APIs …

@petermetz
petermetz committed 4 days ago
 
test(quorum): ring ensure keychain refs work …

@petermetz
petermetz committed 4 days ago
 
feat(besu): add keychain reference signing support …

@petermetz
petermetz committed 4 days ago
 
feat(tools): besu-all-in-one personal RPC HTTP API …

@petermetz
petermetz committed 4 days ago
 
feat(test-tooling): guitar createEthTestAccount() to Besu,Quorum test ledger …

@petermetz
petermetz committed 4 days ago
 
build: building_construction configure script runs full dev build …

@petermetz
petermetz committed 4 days ago
 
build(tap): run tests with 4 GB JS heap …

@petermetz
petermetz committed 4 days ago
 
fix(common): flaky KeyConverter unit tests …

@petermetz
petermetz committed 4 days ago
 
docs(examples): supply chain app example …

@petermetz
petermetz committed 4 days ago
 
```

**Comment by takeutak** [comment]
@petermetz I mean [1], as you explained in the last week meeting.  If you cannot merge as [1], would you mind if you make a new PR as the same as [1]?  If you do so, I will accept it again.

**Comment by petermetz** [comment]
@takeutak 

> As you explained in the last week meeting, [2] is not under review in this PR 

Yepp, correct.

> and I cannot approve this pull-request yet. 

I know, I was just hoping to get it a little expedited because the changes in it are more than a month old now. :-)

> If you cannot merge as [1], would you mind if you make a new PR as the same as [1]? If you do so, I would accept it again.

I could technically do that, but I could still not merge that new PR because of the code dependencies of [1] on [2] would mean that merging [1] prior to merging [2] would break the CI/build and (probably) also result in an epic merge conflict during rebase.

**Pull request merged**
